packages/langium/src/grammar/langium-grammar-semantic-token-provider.ts

```diff
@@ -0,0 +1,32 @@
+/******************************************************************************
+ * Copyright 2022 TypeFox GmbH
+ * This program and the accompanying materials are made available under the
+ * terms of the MIT License, which is available in the project root.
+ ******************************************************************************/
+
+import { SemanticTokenTypes } from 'vscode-languageserver';
+import { AbstractSemanticTokenProvider } from '../lsp/semantic-token-provider';
+import { AstNode } from '../syntax-tree';
+import { isAction, isAssignment, isParameter, isParameterReference, isParserRule, isTerminalRule } from './generated/ast';
+import { isDataTypeRule } from './grammar-util';
+
+export class LangiumGrammarSemanticTokenProvider extends AbstractSemanticTokenProvider {
+
+    protected highlightElement(node: AstNode): boolean {
+        if (isAssignment(node)) {
+            this.highlightFeature(node, 'feature', SemanticTokenTypes.property);
```

Comment:
Since the acceptor isn't actually used inside of the `highlightElement` method, I have another idea: Make `SemanticTokenAcceptor` into an interface:

```
interface SemanticTokenAcceptor {
  highlightToken(line: number, char: number, length: number, type: string, modifiers?: string[]): void
  highlightFeature(node: N, feature: Properties<N>, type: string, ...modifiers: string[]): void
  highlightKeyword(node: AstNode, keyword: string, type: string, ...modifiers: string[]): void
  highlightNode(node: CstNode, type: string, ...modifiers: string[]): void
}
```

That way, we don't have to rely so much on the protected methods of the `AbstractSemanticTokenProvider` and can make better use of the acceptor. Alternatively, the acceptor stays as a single function, but with a lot of overloading and without the `highlightKeyword` method. What do you think?